### ISSUE TITLE
"Fade" option, w/ no background = ugly result + incorrect active classes

### ISSUE THREAD

**Issue by toobulo** [kenwheeler/slick#365]
If you use the slider with the "Fade" option with text only (no images / backgrounds), you notice how the previous slide just drops the opacity after the newest one has faded in.  The problem with this is you can't create a smooth transition (crossFade)using straight CSS opacity transitions because the animation starts late.  

Trying to play with the "slick-active" class is basically useless in the "Fade" mode since the class doesn't exist when it's the first or last item.  

My only thought would be to take advantage of onBeforeChange and addClass (with opacity set to 0) to current slide before it changes.  Could someone provide some insight / workaround I've been able to use the callbacks to pull data from,  but I can't target a current item.

**Comment by wgstjf** [comment]
In case anyone is still having this issue, this is how we fixed it :-)
```php
// On before slide change
$('.your-element').on('beforeChange', function(event, slick, currentSlide, nextSlide){
  $('.your-element[data-slick-index="' + currentSlide + '"]').fadeTo('fast',0);
});
}
```

Cheers,

Will

**Comment by skizzo** [comment]
Sorry, but why is this issue closed? In version 1.5.6 fading two pure black images on white background still creates gray areas..
@wgstjf this wasn't fixing that issue either..
any ideas?

**Comment by Jomar4444** [comment]
@skizzo Hey bro, the solution @wgstjf was correctm all though there are some syntax errors in it. Here's the correct code:

```
// On before slide change
$('.your-element').on('beforeChange', function(event, slick, currentSlide, nextSlide){
  $('.your-element [data-slick-index="' + currentSlide + '"]').fadeTo('fast',0);
});
```
